buildroot-external/patches/occu/0154-WebUI-DynamicResizing.patch

```diff
@@ -0,0 +1,282 @@
+@@ -66,9 +45,6 @@
+     </table>
+ 
+     <script type="text/javascript">
+-      Event.observe(window, "resize", function(event) { recalcDivs(); });
+-      recalcDivs();
+-
+       gotoLoginPage = function() {
+         location.href = "/login.htm";
+       };
+--- occu/WebUI/www/webui/style.css.orig
++++ occu/WebUI/www/webui/style.css
+@@ -100,6 +100,8 @@
+ 
+ html,body
+ {
++  width: 100vw;
++  height: 100vh;
+   margin:0;
+   padding:0;
+   color: $_(textColorB); /* black */
+@@ -134,9 +136,28 @@
+  * * *   Element IDs                     * * *
+  * * * * * * * * * * * * * * * * * * * * * * */
+ 
++/* Login & Logout page */
++
++#maintable {
++  width: 100vw;
++  height: 100vh;
++  border-collapse: collapse;  
++}
++
++#maintable tbody > tr > td {
++  padding: 0;
++}
++
++#maintable #content {
++  height: 100%;
++  background-repeat: no-repeat;
++  background-position: 480px center;
++}
++
+ /* Navigation-Leiste */
+ #navbar 
+ {
++  height: 72px;
+   margin: 0px;
+   background-color: $_(background); /* #183473 */
+   width:100%;
+@@ -144,6 +165,10 @@
+ }
+  
+ /* Haupt-Fenster */
++.Layer0 {
++  height: 100%;
++}
++
+ #_content
+ {
+   height: 70%;
+@@ -177,6 +202,8 @@
+     align-items: flex-start;
+     /* Prevent chrome from outlining this by focus */
+     outline: none;
++    /* Stretch to max available space */
++    height: calc(100% - 168px); /* FIX: header + menubar + footer: 72px + 41px + 55px = 168px */
+ }
+ 
+ #prgcontent
+@@ -196,7 +223,7 @@
+ 
+ /* Kopfzeile */
+ #header {
+-  height: 15%;
++  height: 72px;
+   background-color: $_(background);  /* #183473 */
+   /* background-color: #dcdcdc; */
+   color: $_(textColorB); /* black */
+@@ -206,7 +233,7 @@
+ 
+ /* Fusszeile */
+ #footer {
+-  height:15%;
++  height: 50px;
+   background-color: $_(background); /* #183473 */
+   /* background-color: #dcdcdc; */
+   color: $_(textColorB); /* black */
+@@ -254,12 +281,14 @@
+ /* Linker Bereich der Hauptansicht (falls in zwei Spalten getrennt) */
+ #contentLeft 
+ {
++  height: 100%;
+   vertical-align: top;
+ }
+ 
+ /* Rechter Bereich der Hauptansicht (falls in zwei Spalten getrennt) */
+ #contentRight 
+ { 
++  height: 100%;
+   border-left: solid 5px $_(navBarSeparator); /* #ffffff */ /* Trennlinie zwischen Navigation und Hauptbereich */
+   /* padding-top: 10px; */
+   vertical-align: top;
+@@ -2551,7 +2580,7 @@
+ }
+ #menubar {
+   height: 34px;
+-  width: 100%;
++  /* width: 100%; */
+   border-bottom:solid 5px $_(navBarSeparator); /* #ffffff */
+   background-color: $_(background); /* #183473 */
+   z-index: 189;
+@@ -6274,6 +6303,7 @@
+ }
+ 
+ .CLASS03901 {
++  height: 100%;
+ 	width:100%;
+ }
+ 
+@@ -6335,6 +6365,7 @@
+ }
+ 
+ .CLASS04001 {
++  height: 100%;
+ 	width:100%;
+ }
+ 
+--- occu/WebUI/www/webui/webui.js.orig
++++ occu/WebUI/www/webui/webui.js
+@@ -6942,6 +6942,8 @@
+    */
+   onResize: function()
+   {
++    /**
++     * not required any more, since we moved to css styles
```

Comment:
If really not required, please remove completly! And any chance to get the whole `onResize()` function and the event handler binding eliminated as well?

Reply:
done, see 9a27326